tools/ci/checks/run_pre_commit.sh

```diff
@@ -1,10 +1,32 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-DOCKER_TAG=${DOCKER_TAG:-latest}
-DOCKER_IMAGE=${DOCKER_IMAGE:-mesosphere/dcos-commons:${DOCKER_TAG}}
+set -euxo pipefail
 
-docker run --rm -t \
-    -v $(pwd):/build:ro \
-    -w /build \
-        ${DOCKER_IMAGE} \
-            pre-commit run "$@"
+SCRIPT_DIRECTORY="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+DCOS_COMMONS_DIRECTORY="$(cd "${SCRIPT_DIRECTORY}/../../.." && pwd)"
+
+function was_run_from_submodule () {
+  ! (cd "${DCOS_COMMONS_DIRECTORY}" && test -d .git)
```

Comment:
maybe one line explanation that why this works and not `git rev-parse --is-inside-work-tree` because of `GIT_WORK_TREE` 

as looking at code it might give the impression that we are just checking root directory and checking if its a git project or not

Reply:
The explanation here is that `.git` in checked out submodules is a file whose contents is a git ref. In contrast to the directory with git objects in non-submodules.

I suspect `git rev-parse --is-inside-work-tree` would work here as well (I didn't know about it!), so let me confirm that the result will be the same, and if it is I'll consider using it instead, or in addition.